test/check_compilers/helper.h

```diff
@@ -0,0 +1,37 @@
+#include <iostream>
+#include <type_traits>
+
+// a little helper to pretty-print _multiple_ static_assert failures
+namespace helper {
+
+// intentionally incomplete so it'll show up in the error message with its template args
+template <typename...> struct fail_with;
+
+template <typename Tag, bool B> struct named {
+  // this will be condition
+  static constexpr bool value = B;
+  // this will be the "name" of it so we can recognize the static_assert in the error message
+  using tag = Tag;
+};
+
+template <typename... Ns> inline void static_assert_all(Ns...) {
+  constexpr bool all_ok = (Ns::value && ...);
+  // bcs it's constexpr, this will be dropped by compiler if it's all fine
+  if constexpr (!all_ok) {
```

Comment:
this is quite nice: the arguments here are `struct named`s whose values are what we want to statically assert. if `all_ok` fails, the `tag`s are passed into an incomplete `fail_with<>` struct as arguments, which will trigger an error on instantiation and print the `tag`s in the error. 

so this makes it possible to statically assert many conditions without failing at the first.